### PR TITLE
[monitoring-kubernetes] Switch to CustomResourceStateMetrics for VPA and bump kube-state-metrics

### PR DESCRIPTION
**Pull request by scaps1**
## Description
This PR updates Kubestate Metrics to version 2.13. Since support for VPA metrics was removed in version 2.9, these metrics have been re-integrated using the custom metrics mechanism to ensure the continued functionality of alerts and dashboards. Additionally, relabeling has been added to maintain backward compatibility with the original metric names, avoiding disruptions in dependent systems and configurations.

## Why do we need it, and what problem does it solve?
The update to Kubestate Metrics 2.13 allows us to take advantage of new metrics and improvements, keeping our observability stack current. Introducing custom metrics for the deprecated VPA metrics ensures that we maintain critical alerting and monitoring without service interruption. The relabeling strategy ensures backward compatibility, preserving the previous metric names and minimizing any breaking changes in existing configurations and dashboards. This update is essential for staying up-to-date with the latest metrics and maintaining operational compatibility.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

```changes
section: monitoring-kubernetes
type: chore
summary: Update kube-state-metrics to 2.13
impact_level: default
```

### CODE REVIEW

File: modules/340-monitoring-kubernetes/templates/kube-state-metrics/additional-scrape-config.yaml [hunk 1]
```diff
@@ -16,14 +16,20 @@ spec:
...
+    - __name__
+    targetLabel: __name__
   relabelings:
     - regex: endpoint|namespace|pod|service
       action: labeldrop
     - targetLabel: scrape_endpoint
       replacement: main
     - targetLabel: job
       replacement: kube-state-metrics
-  metricsPath: '/main/metrics'
```
Comment (vladimirGuryanov) [inline modules/340-monitoring-kubernetes/templates/kube-state-metrics/additional-scrape-config.yaml:26]: I guess this was removed by mistake

**Review by vladimirGuryanov — CHANGES_REQUESTED** [review]

File: modules/340-monitoring-kubernetes/templates/kube-state-metrics/additional-scrape-config.yaml (hunk 1 above)
Comment (scaps1) [reply modules/340-monitoring-kubernetes/templates/kube-state-metrics/additional-scrape-config.yaml]: Yes you are right.
Fixed.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vladimirGuryanov. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @scaps1. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @scaps1. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vladimirGuryanov. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/340-monitoring-kubernetes/templates/kube-state-metrics/additional-scrape-config.yaml [hunk 2]
```diff
@@ -16,6 +16,23 @@ spec:
...
+  - sourceLabels: [__name__]
+    regex: kube_verticalpodautoscaler_labels
+    targetLabel: name
+    replacement: ''
+  - action: labeldrop
+    regex: customresource_version
+  - action: labeldrop
+    regex: customresource_kind
+  - action: labeldrop
+    regex: customresource_group
```
Comment (nabokihms) [inline modules/340-monitoring-kubernetes/templates/kube-state-metrics/additional-scrape-config.yaml:35]:
```suggestion
  - action: labeldrop
    regex: customresource_version|customresource_kind|customresource_group
```

**Review by nabokihms — APPROVED** [review]

File: modules/340-monitoring-kubernetes/templates/kube-state-metrics/additional-scrape-config.yaml [hunk 3]
```diff
@@ -16,6 +16,23 @@ spec:
...
+  - action: replace
+    regex: kube_verticalpodautoscaler_labels_info
+    replacement: kube_verticalpodautoscaler_labels
+    sourceLabels:
+    - __name__
+    targetLabel: __name__
+  - sourceLabels: [__name__]
+    regex: kube_verticalpodautoscaler_labels
+    targetLabel: name
+    replacement: ''
```
Comment (lazovskiy) [inline modules/340-monitoring-kubernetes/templates/kube-state-metrics/additional-scrape-config.yaml:29]: I would suggest using the same pattern for readability:
* action is not required here
* keys appear in a different order
* `sourceLabels` written using different yaml-syntax

**Pull request merged**